renderer/plugin.js

```diff
@@ -115,7 +115,34 @@ org.ekstep.questionunitmtf.RendererPlugin = org.ekstep.contentrenderer.questionU
     }
   },
   logTelemetryItemResponse: function (data) {
-    QSTelemetryLogger.logEvent(QSTelemetryLogger.EVENT_TYPES.RESPONSE, {"type": "INPUT", "values": data});
+    QSTelemetryLogger.logEvent(QSTelemetryLogger.EVENT_TYPES.RESPONSE, { "type": "INPUT", "values": data });
+  },
+  /**
+   * shuffles the options array
+   */
+  shuffleOptions: function (options) {
+    var shuffled = [];
```

Comment:
By using simple logic with _.reject & _.shuffle we can achieve this.

var arr = [1,2,3,4,5]
var shArr = _.shuffle(arr)
_.reject(arr, function(val, i){return val != shArr[i]})

if _.reject array length is 0, then all options are mismatched. Else do _shuffle again & try

Reply:
var arr = [1,2,3,4,5] => undefined
_.shuffle(arr) =>(5) [2, 3, 5, 4, 1]
_.shuffle(arr) => (5) [2, 5, 4, 1, 3]

var shArr = _.shuffle(arr) => undefined
shArr => (5) [1, 5, 4, 3, 2]
_.reject(arr, function(val, i){return val != shArr[i]}) => [1]
var shArr = _.shuffle(arr) => undefined
_.reject(arr, function(val, i){return val != shArr[i]}) => []

Reply:
@vinukumar-vs I actually compared the logic I wrote against the `_.shuffle()` and I found that the chances for `_.shuffle()` to return array with elements in the same index is very high. So, there is no guarantee that we will get the desired outcome in 1 or 2 iterations. 

- 

-